api/src/controllers/auth.ts

```diff
@@ -253,11 +253,20 @@ router.get(
 			accountability: accountability,
 			schema: req.schema,
 		});
+		const userService = new UsersService({ schema: req.schema });
 
 		const email = getEmailFromProfile(req.params.provider, req.session.grant.response?.profile);
 
 		req.session?.destroy(() => {});
 
+		const whitelistedDomains = process.env.OAUTH_USER_CREATION_DOMAIN_WHITELIST?.split(",").map(d => d.trim());
+		const emailDomainIsWhitelisted = whitelistedDomains?.some(d => email.endsWith(d)) ?? false;
```

Comment:
I think you could validate domains in whitelist for starting with `@`, otherwise if my whitelist would be like `bar.com`, I can simply bypass this condition with creating own `foobar.com` domain

Reply:
Nice catch! Thanks 🙏